src/cli/cli.hpp

```diff
@@ -0,0 +1,23 @@
+#ifndef __CLI_HPP__
+#define __CLI_HPP__
+
+#include "task.hpp"
+
+
+typedef const struct CLI_menu_
+{
+    char* cmd;
+    const char *const fnName; // menu function name
+    void (*fn)(void); // linked menu function
+} CLI_menu_t;
+
+class CLI : public Task {
+    public:
+    void init(void);
+    STATES_e run(void);
+    void exit(void);
+    CLI_menu_t* CLI_findCommand(char *cmd);
```

Comment:
We are missing docstrings for all of these functions.  I still need to write this up, but I think the standard will be Doxygen formatted docstrings primarily at the symbol declaration focused on usage.  Additional Doxygen formatted docstrings can be at the symbol definition with emphasis on what is going on under the hood.

Reply:
Will do